### PR TITLE
Teal refactor

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Part of https://github.com/insightsengineering/teal/issues/731

### CODE REVIEW

File: R/get_rcode.R [hunk 1]
```diff
@@ -62,6 +62,14 @@ get_rcode <- function(datasets = NULL,
                       title = NULL,
                       description = NULL) {
   checkmate::assert_class(datasets, "FilteredData", null.ok = TRUE)
+
+  lifecycle::deprecate_warn(
+    when = "0.11.2",
```
Comment (nikolas-burkoff) [inline R/get_rcode.R:67]: is this right? Also check others in this PR

Comment (gogonzo) [reply R/get_rcode.R:67]: I've updated this version here

File: vignettes/adding-support-for-reporting.Rmd [hunk 2]
```diff
@@ -271,9 +287,9 @@ example_reporter_module <- function(label = "Example") {
             "verbatim"
           )
           card$append_text("Module Table", "header3")
-          card$append_table(table_r())
+          card$append_table(table_q()[["result"]])
           card$append_text("Show R Code", "header3")
-          card$append_text(paste(teal.code::chunks_get_rcode(simple_chunks), collapse = "\n"), "verbatim")
+          card$append_text(paste(teal.code::get_code(table_q()), collapse = "\n"), "verbatim")
```
Comment (nikolas-burkoff) [inline vignettes/adding-support-for-reporting.Rmd:292]: should this be append src @Polkas

File: vignettes/teal.Rmd [hunk 3]
```diff
@@ -1,7 +1,7 @@
 ---
 title: "Getting Started with Teal"
-author: "Nikolas Burkoff"
-date: "2022-03-24"
+author: "NEST CoreDev"
+date: "2022-07-13"
```
Comment (nikolas-burkoff) [inline vignettes/teal.Rmd:4]:
```suggestion
date: "2022-11-03"
```

File: tests/testthat/test-module_nested_tabs.R [hunk 4]
```diff
@@ -257,3 +257,89 @@ testthat::test_that("srv_nested_tabs.teal_module passes filter_panel_api to the
...
+      d1 = list(dataset = d1, keys = "id", metadata = list("A" = 1)),
+      d2 = list(dataset = d2, keys = "id")
+    ),
+    join_keys = teal.data::join_keys(teal.data::join_key("d1", "d2", c("pk" = "id"))),
+    code = cc,
+    check = TRUE
+  )
+}
+
+testthat::test_that(".datasets_to_data returns filtered data", {
```
Comment (nikolas-burkoff) [inline tests/testthat/test-module_nested_tabs.R:281]:
```suggestion
testthat::test_that(".datasets_to_data returns data which is filtered", {
```

File: vignettes/creating-custom-modules.Rmd [hunk 5]
```diff
@@ -102,52 +108,54 @@ ui_histogram_example <- function(id, datasets, histogram_var) {
...
-
-      # add the code for the plot into the chunks
-      teal.code::chunks_push(
-        bquote(hist(.(as.name(dataname()))[, .(selected())]))
-      )
-
-      # evaluate the chunks
-      teal.code::chunks_safe_eval()
+      # take the filtered data from the data object and add it into the qenv environment
+      teal.code::new_qenv(tdata2env(data), code = get_code(data)) %>%
```
Comment (nikolas-burkoff) [inline vignettes/creating-custom-modules.Rmd:137]:
```suggestion
      teal.code::new_qenv(tdata2env(data), code = get_code_tdata(data)) %>%
```

**Review by nikolas-burkoff — COMMENTED** [review]
Can't approve own PR but I'm happy with this 🏆

File: R/tdata.R [hunk 6]
```diff
@@ -0,0 +1,171 @@
+#' Create a new `tdata` object which contains
```
Comment (donyunardi) [inline R/tdata.R:1]: The title of the manual for tdata is _**Create a new `tdata` object which contains**_.
It felt incorrect.

Along with the description, maybe we can do something like this?
```
#' Create a `tdata` Object
#'
#' Create a new object called `tdata` which contains `data`, a `reactive` list of data.frames
#' (or `MultiAssayExperiment`), with attributes:
#' \itemize{
#'   \item{`code` (`reactive`) containing code used to generate the data}
#'   \item{join_keys (`JoinKeys`) containing the relationships between the data}
#'   \item{metadata (`named list`) containing any metadata associated with the data frames}
#' }
```

It looks better in the manual page:
image

File: vignettes/creating-custom-modules.Rmd [hunk 7]
```diff
@@ -55,18 +58,21 @@ including an encoding panel on the left and main output covering the rest of the
...
   })
 }
 
 ```
 
-The `datasets` argument is a `FilteredData` object which provides a call to extract the data after having been filtered by the filterpanel:
-`datasets$get_data(<<dataname>>, filtered = TRUE)`. For more details about the `FilteredData` object see the supporting `teal` package
-[`teal.slice`](https://insightsengineering.github.io/teal.slice/)
-
+When used inside a teal application called with `teal::init` the `data` argument is a named list of reactive data.frames
```
Comment (donyunardi) [inline vignettes/creating-custom-modules.Rmd:73]:
```suggestion
When used inside a teal application called with `teal::init`, the `data` argument is a named list of reactive data.frames
```

**Review by donyunardi — COMMENTED** [review]
Minor update.

File: R/tdata.R (hunk 6 above)
Comment (nikolas-burkoff) [reply R/tdata.R:1]: yup - done

**Review by gogonzo — APPROVED** [review]

**Pull request merged**